### PR TITLE
Make the API context.Context aware

### LINKED ISSUE #8: Add context.Context support in APIs
First, thanks for this library: the work is outstanding and I really, really love it! Impressive work!

I am integrating Huego (and many other libraries) into a library for my home-automation hub. My goal is to provide a technology-agnostic REST web-service.

A very common pattern for web-services, is to handle properly requests lifetime by honoring the status of the `context.Context` that is attached to `*http.Request` instances. Since this context automatically expires when the client underlying connection closes (or any other fatal network condition), this allows for circuit-breaking: for instance, if I use the `context.Context` from the HTTP request and pass it in an underlying "turn this light on" operation, and the initiating request closes early, it gives a chance to the program to avoid doing unnecessary work by also stopping the "turn this light on" operation sooner.

It is as simple as doing:

```
req = req.WithContext(ctx)
```

Before issuing a HTTP request.

The current Huego APIs don't expose a way of passing in a `context.Context` for each operation. Is that something you'd be interested in? I can probably make a pull-request for it, but I figured I'd discuss it with you first.

There is a way to do that without breaking the API by "overloading" each operation method with a `Ctx` suffix. Say:

```
// Context creation: can be done in many ways.
ctx, cancel := context.WithTimeout(context.Background(), time.Second)
defer cancel()

// Example call.
light.OnCtx(ctx) // Guaranteed to complete when the operation is over or one second passes by, whichever happens first.
```

Please let me know what you think of this.

### PR DESCRIPTION
**Pull request by dlsniper**
This closes #8 while keeping backwards compatibility with the existing API.

I took the liberty of correcting a few spelling errors in the process, but added them to the same commit, hope that's ok.

### CODE REVIEW

File: bridge.go [hunk 1]
```diff
@@ -48,6 +49,11 @@ func (b *Bridge) Login(u string) *Bridge {
 
 // GetConfig returns the bridge configuration
 func (b *Bridge) GetConfig() (*Config, error) {
+	return b.GetConfigWithContext(context.Background())
+}
+
+// GetConfigContext returns the bridge configuration
+func (b *Bridge) GetConfigWithContext(ctx context.Context) (*Config, error) {
```
Comment (amimof) [inline bridge.go:56]: Remame this to `GetConfigContext`

**Comment by dlsniper** [comment]
> Thanks for this. Awsome work 👏 

Thank you!

> Please add unit tests for the new functions

How could we do this efficiently? I'd rather not duplicate the existing tests to just add a new parameter to them. The existing tests will call both the "old" and "new" functions as well.

**Comment by amimof** [comment]
@dlsniper I'm working on this PR #20 that adds better coverage reports using https://codecov.io. Once that's merged we can better see what isn't covered here.

**Comment by amimof** [comment]
@dlsniper Just checked and coverage is ok, no need to add any tests. I'll merge once the review is resolved.

**Pull request merged**